scripts/reduce_CG1D.py

```diff
@@ -112,12 +108,13 @@ def main(inputfile: Union[str, Path], outputdir: Union[str, Path]) -> int:
     save_config(config_dict, config_fn)
 
     # step 4: call the auto reduction with updated dict
-    workflow = WorkflowEngineAuto(config_dict)
     try:
+        workflow = WorkflowEngineAuto(config_dict)
         workflow.run()
         return WORKFLOW_SUCCESS
-    except WorkflowEngineError:
-        return WORKFLOW_ERROR
+    except WorkflowEngineError as e:
+        logger.exception("Failed to create and run workflow")
```

Comment:
can you also include `str(e)` in the log message?

Reply:
`logger.exception` prints the message supplied, then the full stack trace

Reply:
not that I don't believe you but [I had to check](https://docs.python.org/3/library/logging.html#logging.exception) this hidden behavior :smile: